spec/02-integration/13-vaults/05-ttl_spec.lua

```diff
@@ -189,36 +193,52 @@ describe("vault ttl and rotation (#" .. strategy .. ") #" .. vault.name, functio
     helpers.unsetenv("KONG_LUA_PATH_OVERRIDE")
   end)
 
+  local function check_plugin_secret(expect, ttl, leeway)
+    leeway = leeway or 0.25 -- 25%
+
+    local timeout = ttl + (ttl * leeway)
+
+    assert
+            .with_timeout(timeout)
+            .with_step(0.5)
+            .eventually(function()
+      local res = http_get("/")
+      local value
+      if expect == "" then
+        value = res.headers[DUMMY_HEADER] or ""
+        if value == "" then
+          return true
+        end
+
+      else
+        value = assert.response(res).has.header(DUMMY_HEADER)
+        if value == expect then
+          return true
+        end
+      end
+
+      return nil, { expected = expect, got = value }
+    end)
+    .is_truthy("expected plugin secret to be updated to '" .. tostring(expect) .. "' "
+            .. "within " .. tostring(timeout) .. " seconds")
+  end
 
   it("updates plugin config references (backend: #" .. vault.name .. ")", function()
-    local function check_plugin_secret(expect, ttl, leeway)
-      leeway = leeway or 0.25 -- 25%
-
-      local timeout = ttl + (ttl * leeway)
-
-      assert
-        .with_timeout(timeout)
-        .with_step(0.5)
-        .eventually(function()
-          local res = http_get("/")
-          local value = assert.response(res).has.header(DUMMY_HEADER)
-
-          if value == expect then
-            return true
-          end
-
-          return nil, { expected = expect, got = value }
-        end)
-        .is_truthy("expected plugin secret to be updated to '" .. expect .. "' "
-                .. "' within " .. tostring(timeout) .. "seconds")
-    end
-
-    vault:update_secret(secret, "old", { ttl = 5 })
+    vault:update_secret(secret, "old", { ttl = 2, resurrect_ttl = 2 })
     check_plugin_secret("old", 5)
-
-    vault:update_secret(secret, "new", { ttl = 5 })
+    vault:update_secret(secret, "new", { ttl = 2, resurrect_ttl = 2 })
```

Comment:
What I wrote here -> https://github.com/Kong/kong/pull/11815/files#r1371755481

Lets have separate tests. What about a separate file that is very close to this one but uses `ttl AND resurrect_ttl` ?

Reply:
There is now a separate file.